tests/console/ipsec_tools_h2h.pm

```diff
@@ -0,0 +1,92 @@
+# SUSE's Racoon tests
+#
+# Copyright © 2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: Test Racoon host-to-host scenario
+# Maintainer: Jiawei Sun <jiawei.sun@suse.com>
+
+use strict;
+use base "consoletest";
+use lockapi;
+use testapi;
+use utils;
+use mmapi;
+use mm_network;
+
+sub run {
+    my $self = shift;
+    select_console 'root-console';
+    my $is_ipsec_primary = get_var('IPSEC_PRIMARY');
+
+    # Static IP address for Primary host and secondary host
+    my $ipsec_primary_ip   = "10.0.2.1/24";
+    my $ipsec_secondary_ip = "10.0.2.50/24";
+    my $my_static_ip       = $is_ipsec_primary ? $ipsec_primary_ip : $ipsec_secondary_ip;
+
+    # install ipsec-tools and config gateway and static ip on worker
+    zypper_call '-n in ipsec-tools';
+    configure_default_gateway;
+    configure_static_ip($my_static_ip);
+
+    if ($is_ipsec_primary) {
+        set_config($is_ipsec_primary, $ipsec_primary_ip, $ipsec_secondary_ip);
+        my $children = get_children();
+        my $child_id = (keys %$children)[0];
+        mutex_create('ipsec_primary');
+    }
+    else {
+        set_config($is_ipsec_primary, $ipsec_secondary_ip, $ipsec_primary_ip);
+        mutex_lock('ipsec_primary');
+        mutex_unlock('ipsec_primary');
+        my ($primary) = split('/', $ipsec_primary_ip);
+
+        # Test the secure channel, ping primary from secondary
+        assert_script_run "ping -c 6 $primary";
+    }
+}
+
+# Get pre-generated certs and config template for primary and secondary
+
+sub set_config {
+    my ($is_primary, $host_ip, $remote_ip) = @_;
+
+    # Split subnet mask
+    my ($host, $mask) = split('/', $host_ip);
+    my ($remote) = split('/', $remote_ip);
+    my $cert = $is_primary ? "server" : "client";
+
+    # Download pre-generated certs
+    my $setup_script = "curl -f -v " . autoinst_url . "/data/openssl/ca-cert.pem > /etc/racoon/cert/ca-cert.pem\n";
+    $setup_script .= "curl -f -v " . autoinst_url . "/data/openssl/$cert.pem > /etc/racoon/cert/$cert-cert.pem\n";
```

Comment:
It's not the problem with code here, but in some architectures like s390x and ppc64 at the moment we have issues when typing long strings. If you're going to replace "script_output" with assert_script_run it would be better for stability to run each line using script_run method.